NuKeeper/RepositoryInspection/SolutionsRestore.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+using NuKeeper.NuGet.Process;
+
+namespace NuKeeper.RepositoryInspection
+{
+    public class SolutionsRestore
+    {
+        private readonly ISolutionRestoreCommand _solutionRestoreCommand;
+
+        public SolutionsRestore(ISolutionRestoreCommand solutionRestoreCommand)
+        {
+            _solutionRestoreCommand = solutionRestoreCommand;
+        }
+
+        public async Task Restore(string rootDirectory)
+        {
+            var solutionFiles = SolutionFinder.FindSolutions(rootDirectory);
+
+            foreach (var solutionPath in solutionFiles)
+            {
+                await _solutionRestoreCommand.Restore(solutionPath);
```

Comment:
I would expect to read `_command.Invoke(solutionPath);` here - doing it this way leaves a lot of openings for bundling up many different commands into this ISolutionRestoreCommand object, which would be an ugly solution